packages/suite/src/middlewares/firmware/firmwareMiddleware.ts

```diff
@@ -11,12 +11,13 @@ import { FIRMWARE } from '@suite/actions/firmware/constants';
 const firmware = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) => (
     action: Action,
 ): Action => {
-    const prevApp = api.getState().router.app;
+    const { app: prevApp } = api.getState().router;
 
     // pass action
     next(action);
 
     const { status } = api.getState().firmware;
+    const { app } = api.getState().router;
```

Comment:
I'm not sure here, but execution of the `action` is async, right? So this `getState()` call could be executed even before the `action` is processed so the `app` could be the same as `prevApp`, right? 

In that case if it is ok, then you probably don't need to define it twice and if it's not ok, then we need to handle it differently.

Reply:
Yes, you are definitely right. 

I need only "previous" `app` here because when you have firmware modal open and you disconnect your device, it opens another modal (device disconnected), so the `firmware` one is the "previous one".